napari/_qt/_qapp_model/qactions/__init__.py

```diff
@@ -84,3 +88,21 @@ def init_qactions() -> None:
     app.menus.append_menu_items(
         chain(FILE_SUBMENUS, VIEW_SUBMENUS, DEBUG_SUBMENUS, LAYERS_SUBMENUS)
     )
+
+
+def add_dummy_actions(context: Context) -> None:
+    from napari._app_model import get_app
+    from napari._app_model.constants._menus import MenuId
+    from napari._app_model.utils import get_dummy_action, is_empty_menu
+
+    app = get_app()
+
+    actions = []
+    for menu_id in MenuId.contributables():
+        # assumes leaves are unique!!
```

Comment:
maybe add a bit more comments here — and a docstring to the function! — about what you're doing with the keys? I don't actually grok it — each menu menu has its own dummy entry? And that key gets added/removed when the menu is opened? Or it gets shown/hidden? How does that work? Comments could make the whole mechanism clearer at a glance...

Reply:
Expanded this to explain a bit more about how it works. Let me know if anything is still unclear!